server-spi/src/main/java/org/keycloak/models/RealmModel.java

```diff
@@ -862,6 +862,15 @@ default List<GroupModel> searchForGroupByName(String search, Integer first, Inte
      */
     Stream<GroupModel> searchForGroupByNameStream(String search, Integer first, Integer max);
 
+    /**
+     * Searches for groups by provided group attributes. Results that match the given filter are returned as a stream.
+     * @param attributes {@code String} Name of a group to be used as a filter.
+     * @param first {@code Integer} Index of the first desired group. Ignored if negative or {@code null}.
+     * @param max {@code Integer} Maximum number of returned groups. Ignored if negative or {@code null}.
+     * @return Stream of {@link GroupModel}. Never returns {@code null}.
+     */
+    Stream<GroupModel> searchGroupsByAttributes(Map<String, String> attributes, Integer first, Integer max);
```

Comment:
Do we need this method? I am aware, that we have quite a lot of similar methods in the `RealmModel` interface, however, we should aim for having only methods relevant to realms in `RealmModel` and avoid such shortcut methods in favour of `session.groups().****`.

Reply:
I think this was done to ease the process of collecting all of the attributes and checking that they are configured to be searchable before executing the query. Also to keep parity with `searchClientsByAttribute`

Reply:
Not sure what you mean by:

> ease the process of collecting all of the attributes and checking that they are configured to be searchable before executing the query

Can you be more specific? It seems that call:
```
realm.searchGroupsByAttributes(searchedAttributes, first, max)
```
is equivalent to
```
session.groups().searchGroupByAttributes(session, searchedAttributes, first, max)
```

The only difference is there is one more method call in a `RealmModel` implementation in the first case. My comment aims to get rid of this method call if it is possible.

Reply:
We tried to align all the calls from `GroupsResource` in the same way that was implemented for other methods i.e. using `RealmModel` object. If you want, I can remove `searchGroupByAttributes` from `RealmModel` and all its implementation and replace it with keycloak session call that you shared.

Reply:
Removed `searchGroupsByAttributes` from `RealmModel` and its implementations